Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -160,29 +161,40 @@ protected BlockPuller(ConcurrentChain chain, IReadOnlyNodesCollection nodes, Pro
         /// <param name="cancellationToken">Cancellation token to be used by derived classes that allows the caller to cancel the execution of the operation.</param>
         public virtual void BlockPushed(uint256 blockHash, DownloadedBlock downloadedBlock, CancellationToken cancellationToken)
         {
-            this.logger.LogTrace($"(blockHash:'{blockHash}',downloadedBlock.Length:{downloadedBlock.Length})");
+            this.logger.LogTrace($"({nameof(blockHash)}:'{blockHash}',{nameof(downloadedBlock)}.{nameof(downloadedBlock.Length)}:{downloadedBlock.Length})");
             this.logger.LogTrace("(-)");
         }
 
         /// <inheritdoc />
         public void InjectBlock(uint256 blockHash, DownloadedBlock downloadedBlock, CancellationToken cancellationToken)
         {
+            this.logger.LogTrace($"({nameof(blockHash)}:'{blockHash}',{nameof(downloadedBlock)}.{nameof(downloadedBlock.Length)}:{downloadedBlock.Length})");
+
             if (this.AddDownloadedBlock(blockHash, downloadedBlock))
               this.BlockPushed(blockHash, downloadedBlock, cancellationToken);
+
+            this.logger.LogTrace("(-)");
         }
 
         /// <inheritdoc />
         public virtual void AskBlocks(ChainedBlock[] downloadRequests)
         {
-            BlockPullerBehavior[] nodes = GetNodeBehaviors();
+            this.logger.LogTrace($"({nameof(downloadRequests)}:{string.Join(",", downloadRequests.Select(r => r.Height))})");
 
-            Dictionary<int, InventoryVector> vectors = new Dictionary<int, InventoryVector>();
-            foreach (ChainedBlock request in downloadRequests)
+            BlockPullerBehavior[] nodes = this.GetNodeBehaviors();
+            if ((nodes.Length > 0) && (downloadRequests.Length > 0))
             {
-                InventoryVector vector = new InventoryVector(InventoryType.MSG_BLOCK, request.HashBlock);
-                vectors.Add(request.Height, vector);
+                Dictionary<int, InventoryVector> vectors = new Dictionary<int, InventoryVector>();
```

Comment:
yes, that is my faul on code style rule no. 10  (although strictly speaking not really, but yeah, i'll fix it)